services/src/main/java/org/fao/geonet/api/selections/UserSelectionsApi.java

```diff
@@ -244,26 +253,48 @@ List<String> getSelectionRecords(
             HttpSession httpSession
     )
         throws Exception {
-        Optional<Selection> selection = selectionRepository.findById(selectionIdentifier);
-        if (!selection.isPresent()) {
-            throw new ResourceNotFoundException(String.format(
-                "Selection with id '%d' does not exist.",
-                selectionIdentifier
-            ));
-        }
+        UserSession session = ApiUtils.getUserSession(httpSession);
+        Profile myProfile = session.getProfile();
+        String myUserId = session.getUserId();
 
-        Optional<User> user = userRepository.findById(userIdentifier);
-        if (!user.isPresent()) {
-            throw new ResourceNotFoundException(String.format(
-                "User with id '%d' does not exist.",
-                selectionIdentifier
-            ));
-        }
+        if (myProfile.equals(Profile.Administrator) || myProfile.equals(Profile.UserAdmin) ||
+            myUserId.equals(Integer.toString(userIdentifier))) {
+            Optional<Selection> selection = selectionRepository.findById(selectionIdentifier);
+            if (!selection.isPresent()) {
+                throw new ResourceNotFoundException(String.format(
+                    "Selection with id '%d' does not exist.",
+                    selectionIdentifier
+                ));
+            }
 
-        if (selection.isPresent()) {
-            return umsRepository.findMetadata(selectionIdentifier, userIdentifier);
+            Optional<User> user = userRepository.findById(userIdentifier);
+            if (!user.isPresent()) {
+                throw new ResourceNotFoundException(String.format(
+                    "User with id '%d' does not exist.",
+                    selectionIdentifier
+                ));
+            }
+
+            if (!(myUserId.equals(Integer.toString(userIdentifier))) && myProfile == Profile.UserAdmin) {
+
+                //--- retrieve session user groups and check to see whether this user is
+                //--- allowed to get this info
+                List<Integer> adminlist = userGroupRepository.findGroupIds(where(hasUserId(Integer.parseInt(myUserId))).or(hasUserId
+                    (userIdentifier)));
+                if (adminlist.isEmpty()) {
+                    throw new IllegalArgumentException("You don't have rights to do this because the user you want to edit is not part of your group");
+                }
+            }
+
+            if (selection.isPresent()) {
```

Comment:
This check can be removed, previously the code checks `!selection.isPresent()` to throw an exception.